frontend/src/components/cases/case-attachments-section.tsx

```diff
@@ -0,0 +1,735 @@
+        if (body.detail) {
+          errorMessage =
+            typeof body.detail === "string"
+              ? body.detail
+              : body.detail.message || JSON.stringify(body.detail)
+        }
+      }
+
+      toast({
+        title: "Delete failed",
+        description: `Failed to delete attachment. ${errorMessage}`,
+      })
+    },
+  })
+
+  // Add file validation function
+  const validateFile = (file: File): boolean => {
+    if (file.size > MAX_FILE_SIZE_BYTES) {
+      toast({
+        title: "File too large",
+        description: `${file.name} is ${formatFileSize(file.size)}. Maximum file size is 20MB.`,
+        variant: "destructive",
+      })
+      return false
+    }
+    return true
+  }
+
+  const handleFileSelect = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0]
+    if (file) {
+      // Validate file size before uploading
+      if (!validateFile(file)) {
+        return
+      }
+      setIsUploading(true)
+      uploadMutation.mutate(file)
+    }
+    // Reset input
+    if (fileInputRef.current) {
+      fileInputRef.current.value = ""
+    }
+  }
+
+  const handleAddAttachment = () => {
+    fileInputRef.current?.click()
+  }
+
+  const handleDragEnter = useCallback((e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setIsDragOver(true)
+  }, [])
+
+  const handleDragLeave = useCallback((e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setIsDragOver(false)
+  }, [])
+
+  const handleDragOver = useCallback((e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+  }, [])
+
+  const handleDrop = useCallback(
+    (e: React.DragEvent) => {
+      e.preventDefault()
+      e.stopPropagation()
+      setIsDragOver(false)
+
+      const files = Array.from(e.dataTransfer.files)
+      const file = files[0] // Only handle the first file
+
+      if (file) {
+        // Validate file size before uploading
+        if (!validateFile(file)) {
+          return
+        }
+        setIsUploading(true)
+        uploadMutation.mutate(file)
+      }
+    },
+    [uploadMutation]
+  )
+
+  const handleDownload = async (attachment: CaseAttachmentRead) => {
+    try {
+      const response = (await casesDownloadAttachment({
+        caseId,
+        workspaceId,
+        attachmentId: attachment.id,
+      })) as CaseAttachmentDownloadResponse
+
+      // Response now contains presigned URL, not binary data
+      const downloadUrl = response.download_url
+
+      if (!downloadUrl) {
+        throw new Error("No download URL received from server")
+      }
+
+      // Create a hidden link element and trigger download
+      const link = document.createElement("a")
+      link.href = downloadUrl
+      link.download = attachment.file_name
+      link.style.display = "none"
+
+      document.body.appendChild(link)
+      link.click()
+      document.body.removeChild(link)
```

Comment:
should probably clean up with something like:
```typescript
try {
...
} finally {
    a.remove() // Clean up
    window.URL.revokeObjectURL(downloadUrl)
  }
  ```
  see `exportWorkflow` for reference